### PR TITLE
fix: error when lockfile without metadata entry is encountered

### PR DESCRIPTION
**Pull request by iloveitaly**
Without this, you'll get a cryptic 'metadata' in red when running `poetry install`. This can occur if a poetry.lock file is empty

Not sure of the best way to write a test for this, point me in the right direction and I'll write a test.

# Pull Request Check List

- [ ] Added **tests** for changed code.
- [ ] Updated **documentation** for changed code.

### CODE REVIEW

**Comment by iloveitaly** [comment]
added test + fixed linting

File: tests/packages/test_locker.py [hunk 1]
```diff
@@ -622,6 +622,38 @@ def test_reading_lock_file_should_raise_an_error_on_invalid_data(
...
+url = "https://foo.bar"
+reference = "legacy"
+"""
+    with locker.lock.open("w", encoding="utf-8") as f:
+        f.write(content)
+
+    with pytest.raises(RuntimeError) as e:
+        _ = locker.lock_data
+
+    assert (
```
Comment (eamanu) [inline tests/packages/test_locker.py:650]: I'd run the assert in the `with pytest.raises` block.

**Review by eamanu — APPROVED** [review]

File: tests/packages/test_locker.py (hunk 1 above)
Comment (iloveitaly) [reply tests/packages/test_locker.py]: good catch, fixed!

Comment (radoering) [reply tests/packages/test_locker.py]: IMO, only the code that is expected to raise the exception should be in the `with pytest.raises`. That's also how we did it in all other tests, so please revert.

Comment (iloveitaly) [reply tests/packages/test_locker.py]: reverted!

**Review by radoering — APPROVED** [review]

**Pull request merged**